From 9b3410961e52b4666a118158512e050462711fd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 6 Nov 2014 09:56:49 +0000
Subject: [PATCH] Resolves: fdo#76581 copy-and-paste -> slideshow crash in
 presenter console

crash started after...

commit 0218b0e2fa510d7acd7413dbedefd9ad50257f76
Date:   Mon Jul 15 19:45:44 2013 +0300
    fdo#65457 -  Provide visual clues in presenter view.

Change-Id: I19d84800bd5924f2dcc9e5debcf18ef95577105c
---
 sdext/source/presenter/PresenterController.cxx | 28 ++++++++++++++++++--------
 1 file changed, 20 insertions(+), 8 deletions(-)

diff --git a/sdext/source/presenter/PresenterController.cxx b/sdext/source/presenter/PresenterController.cxx
index 43d60ac..6c64bd7 100644
--- a/sdext/source/presenter/PresenterController.cxx
+++ b/sdext/source/presenter/PresenterController.cxx
@@ -358,7 +358,7 @@ void PresenterController::UpdatePaneTitles (void)
                     sCurrentSlideName = sName;
             }
         }
-        catch (beans::UnknownPropertyException&)
+        catch (const beans::UnknownPropertyException&)
         {
         }
     }
@@ -585,10 +585,16 @@ bool PresenterController::HasTransition (Reference<drawing::XDrawPage>& rxPage)
     if( rxPage.is() )
     {
         Reference<beans::XPropertySet> xSlidePropertySet (rxPage, UNO_QUERY);
-        xSlidePropertySet->getPropertyValue("TransitionType") >>= aTransitionType;
-        if( aTransitionType > 0 )
+        try
+        {
+            xSlidePropertySet->getPropertyValue("TransitionType") >>= aTransitionType;
+            if (aTransitionType > 0)
+            {
+                bTransition = true;
+            }
+        }
+        catch (const beans::UnknownPropertyException&)
         {
-            bTransition = true;
         }
     }
     return bTransition;
@@ -597,8 +603,6 @@ bool PresenterController::HasTransition (Reference<drawing::XDrawPage>& rxPage)
 bool PresenterController::HasCustomAnimation (Reference<drawing::XDrawPage>& rxPage)
 {
     bool bCustomAnimation = false;
-    presentation::AnimationEffect aEffect = presentation::AnimationEffect_NONE;
-    presentation::AnimationEffect aTextEffect = presentation::AnimationEffect_NONE;
     if( rxPage.is() )
     {
         sal_uInt32 i, nCount = rxPage->getCount();
@@ -606,8 +610,16 @@ bool PresenterController::HasCustomAnimation (Reference<drawing::XDrawPage>& rxP
         {
             Reference<drawing::XShape> xShape(rxPage->getByIndex(i), UNO_QUERY);
             Reference<beans::XPropertySet> xShapePropertySet(xShape, UNO_QUERY);
-            xShapePropertySet->getPropertyValue("Effect") >>= aEffect;
-            xShapePropertySet->getPropertyValue("TextEffect") >>= aTextEffect;
+            presentation::AnimationEffect aEffect = presentation::AnimationEffect_NONE;
+            presentation::AnimationEffect aTextEffect = presentation::AnimationEffect_NONE;
+            try
+            {
+                xShapePropertySet->getPropertyValue("Effect") >>= aEffect;
+                xShapePropertySet->getPropertyValue("TextEffect") >>= aTextEffect;
+            }
+            catch (const beans::UnknownPropertyException&)
+            {
+            }
             if( aEffect != presentation::AnimationEffect_NONE ||
                 aTextEffect != presentation::AnimationEffect_NONE )
             {
-- 
1.9.3

